include/zephyr/arch/arm64/arm_mmu.h

```diff
@@ -207,6 +207,28 @@ struct arm_mmu_ptables {
 #define MMU_REGION_FLAT_ENTRY(name, adr, sz, attrs) \
 	MMU_REGION_ENTRY(name, adr, adr, sz, attrs)
 
+/**
+ * @brief Auto generate mmu region entry for node_id
+ *
+ * Example usage:
+ *
+ * @code{.c}
+ *      DT_FOREACH_STATUS_OKAY_VARGS(nxp_imx_gpio,
+ *				  MMU_REGION_DT_FLAT_ENTRY,
+ *				 (MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS))
+ * @endcode
+ *
+ * @note  Since devicetree_generated.h does not include
+ *        node_id##_P_reg_FOREACH_PROP_ELEM* definitions,
+ *        we can't automate dts node with multiple reg
+ *        entries.
```

Comment:
I think this can be achieved using `DT_PROP_HAS_IDX` and `LISTIFY`.

Reply:
Oh, didn't realize `LISTIFY` was a thing.

```c
LISTIFY(DT_NUM_REGS(node_id), MMU_REGION_DT_FLAT_ENTRY_BY_IDX, (,), node_id, attrs)
```

should be ok. But there's still a limitation: all regs must have same `attrs`. For GIC, this is true. But for external memory controller such as FlexSPI, they usually require different `attrs`. I think maybe the use case is too limited. 

Given that:
- Most peripheral with multiple instances only requires 1 entry per device.
- Devices such as GIC only have 1 instance per SoC and is always enabled, it's okay to write 2 entries for it.
- Memory controller such as FlexSPI are not supported on MPU yet. They could manage mappings themself with `device_mmio` APIs by the time they are enabled.

I think we can leave it as-is. The purpose of this macro is to conditionally map simple peripherals with lots of instances, making the use of existing MCU-ready driver easier to use on MPU.

Reply:
mmh, alright, let's leave this as is for now.